src/lib/V3DutchDecayLib.sol

```diff
@@ -0,0 +1,145 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+pragma solidity ^0.8.0;
+
+import {OutputToken, InputToken} from "../base/ReactorStructs.sol";
+import {V3DutchOutput, V3DutchInput, V3Decay} from "../lib/V3DutchOrderLib.sol";
+import {FixedPointMathLib} from "solmate/src/utils/FixedPointMathLib.sol";
+import {sub} from "./MathExt.sol";
+import {Uint16Array, fromUnderlying} from "../types/Uint16Array.sol";
+
+/// @notice thrown when the decay curve is invalid
+error InvalidDecayCurve();
+
+/// @notice helpers for handling non-linear dutch order objects
+library V3DutchDecayLib {
```

Comment:
consider naming NonlinearDutchDecayLib for specificity here - generally I see these libraries as independent from the reactor implementations they service, where reactors are just a specific aggregation of features from libraries